sentence_transformers/losses/MatryoshkaLoss.py

```diff
@@ -165,12 +158,13 @@ def __init__(
         dims_weights = zip(matryoshka_dims, matryoshka_weights)
         self.matryoshka_dims, self.matryoshka_weights = zip(*sorted(dims_weights, key=lambda x: x[0], reverse=True))
         self.n_dims_per_step = n_dims_per_step
+        self.cached_losses = (CachedMultipleNegativesRankingLoss, CachedGISTEmbedLoss, CachedMultipleNegativesSymmetricRankingLoss)
```

Comment:
Nice - this has a niche but useful bonus effect that people interested in creating their own Cached loss can extend this after initializing the loss.

E.g. 
```python
loss = CachedCustomLoss(model, ...)
loss = MatryoshkaLoss(model, loss, matryoshka_dims=[768, 512, 256, 128, 64, 32])
loss.loss.calculate_loss = CachedLossDecorator(...)
loss.cached_losses = CachedCustomLoss
```